curp/src/server/raw_curp/log.rs

```diff
@@ -483,4 +516,19 @@ mod tests {
             .enumerate()
             .for_each(|(idx, &size)| assert_eq!(size, entry_size * idx.numeric_cast::<u64>()));
     }
+
+    #[test]
+    fn compact_test() {
+        let (log_tx, _log_rx) = mpsc::unbounded_channel();
+        let mut log = Log::<TestCommand>::new(log_tx, default_batch_max_size(), 10);
+
+        for _ in 0..30 {
+            log.push_cmd(0, Arc::new(TestCommand::default())).unwrap();
+        }
+        log.last_as = 22;
+        log.last_exe = 22;
+        log.compact();
+        assert_eq!(log.base_index, 12);
+        assert_eq!(log.entries.front().unwrap().index, 13);
```

Comment:
The `compact` method will compact not only log entries but also `batch_index`. Make sure that `compact_test` will cover `batch_index`.